docs/resources/service_principal.md

```diff
@@ -0,0 +1,64 @@
+# databricks_service_principal Resource
+
+Directly creates service principal, that could be added to [databricks_group](group.md) within workspace.
+
+## Example Usage
+
+Creating regular service principal:
+
+```hcl
+resource "databricks_service_principal" "me" {
+  application_id = "00000000-0000-0000-0000-000000000000"
+}
+```
+
+Creating service principal with administrative permissions - referencing special `admins` [databricks_group](../data-sources/group.md) in [databricks_group_member](group_member.md) resource:
+
+```hcl
+data "databricks_group" "admins" {
+    display_name = "admins"
+}
+
+resource "databricks_service_principal" "me" {
+  application_id    = "00000000-0000-0000-0000-000000000000"
```

Comment:
i'm currently running this by our engineering folks to agree on name for this field, because service principal identification might look differently on AWS

Reply:
any suggestions?

Reply:
still internal discussions :) i'll try to run this PR against AWS deployment this week

Reply:
@nfx I removed the code related with Role (seems only an aws concept)from service principal, if you want to keep it just revert this commit: de41db032a1dd281c83e193aedce3f7c8ee80643

Reply:
@nfx  any updates?